jobsdb/jobsdb.go

```diff
@@ -699,14 +710,15 @@ multiple users of JobsDB
 dsRetentionPeriod = A DS is not deleted if it has some activity
 in the retention time
 */
-func (jd *HandleT) Setup(ownerType OwnerType, clearAll bool, tablePrefix string, retentionPeriod time.Duration, migrationMode string, registerStatusHandler bool, queryFilterKeys QueryFiltersT) {
+func (jd *HandleT) Setup(ownerType OwnerType, clearAll bool, tablePrefix string, retentionPeriod time.Duration, migrationMode string, registerStatusHandler bool, queryFilterKeys QueryFiltersT, preBackupHandlers []prebackup.Handler) {
```

Comment:
```suggestion
func (jd *HandleT) Setup(ownerType OwnerType, clearAll bool, tablePrefix string, retentionPeriod time.Duration, migrationMode string, registerStatusHandler bool, queryFilterKeys QueryFiltersT) {
```

We don't need to pass `preBackupHandlers` here. `.Setup` is already having too many parameters. 

1. Use New...(), WithPreBackupHandle pattern
2. expose `preBackupHandlers` and inject the handler directly.

Reply:
Why exposing it for initialisation purposes is better than including it in the existing initialiser function?

Currently `jobsdb.HandleT` has none of its dependency fields exported, which is not bad.
Generally speaking, refactoring jobsdb's initialization is indeed needed, but shall we deal with it now?